src/wp-includes/media.php

```diff
@@ -1008,7 +1015,9 @@ function wp_get_attachment_image_src( $attachment_id, $size = 'thumbnail', $icon
  * @param int          $attachment_id Image attachment ID.
  * @param string|int[] $size          Optional. Image size. Accepts any registered image size name, or an array
  *                                    of width and height values in pixels (in that order). Default 'thumbnail'.
- * @param bool         $icon          Optional. Whether the image should be treated as an icon. Default false.
+ * @param array|bool   $args          Optional. Additional arguments for the image to return. For backward
+ *                                    compatibility boolean can be used to determine whether the image should
+ *                                    fall back to a mime type icon. Default false.
```

Comment:
See my above comment on documenting the possible keys more extensively, following core best practices.